fog/view/server/src/block_tracker.rs

```diff
@@ -72,14 +71,33 @@ impl BlockTracker {
 
     /// Given a list of ingestable ranges, missing blocks and current state, calculate the highest
     /// processed block count number. The highest processed block count number is the block count
-    /// for which we know we have loaded all available data.
+    /// for which we know we have loaded all required data, so the users can potentially compute
+    /// their balance up to this block without missing any transactions.
+    ///
+    /// Arguments:
+    /// * highest_known_block_index:
+    ///   The highest block index known to have appeared in the blockchain.
+    ///   If there are no ingress keys at all right now, then this is also the highest
```

Comment:
Isn't the value of this coming from the sql database?
If so, how would there ever be a value for it if there are no ingress keys?

Reply:
you're right, this is kind of a silly way to do this

this function should probably figure out the highest known block index based on what it sees in the ingress keys record set

Reply:
Maybe in the ingress keys list + missing blocks list...?